docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,505 @@
+relevant to your model.
+- Dumb getters and setters: since Java lacks support for properties in the language, we have to create fields,
+then generate getters and setters for those fields, even if they don't actually do anything more than read/write
+the fields.
+- Traditional EE patterns advise to split entity definition (the model) from the operations you can do on them
+(DAOs, Repositories), but really that requires an unnatural split between the state and its operations even though
+we would never do something like that for regular objects in the Object Oriented architecture, where state and methods are in the same class. Moreover, this requires two classes per entity, and requires injection of the DAO or Repository where you need to do entity operations, which breaks your edit flow and requires you to get out of the code you're writing to set up an injection point before coming back to use it.
+- MongoDB queries are super powerful, but overly verbose for common operations, requiring you to write queries even
+when you don't need all the parts.
+- MongoDB queries are JSON based, so you will need some String manipulation or using the `Document` type and it will needs a lot of boilerplate code.
+
+With Panache, we took an opinionated approach to tackle all these problems:
+
+- Make your entities extend `PanacheMongoEntity`: it has an ID field that is auto-generated. If you require
+a custom ID strategy, you can extend `PanacheMongoEntityBase` instead and handle the ID yourself.
+- Use public fields. Get rid of dumb getter and setters. Under the hood, we will generate all getters and setters
+that are missing, and rewrite every access to these fields to use the accessor methods. This way you can still
+write _useful_ accessors when you need them, which will be used even though your entity users still use field accesses.
+- Don't use DAOs or Repositories: put all your entity logic in static methods in your entity class. Your entity superclass comes with lots of super useful static methods and you can add your own in your entity class. Users can just start using your entity `Person` by typing `Person.` and getting completion for all the operations in a single place.
+- Don't write parts of the query that you don't need: write `Person.find("order by name")` or
+`Person.find("name = ?1 and status = ?2", "stef", Status.Alive)` or even better `Person.find("name", "stef")`.
+
+That's all there is to it: with Panache, MongoDBhas never looked so trim and neat.
+
+== Defining entities in external projects or jars
+
+MongoDB with Panacge relies on compile-time bytecode enhancements to your entities. 
+If you define your entities in the same project where you build your Quarkus application, everything will work fine. 
+If the entities come from external projects or jars, you can make sure that your jar is treated like a Quarkus application library by adding an empty `META-INF/beans.xml` file.
```

Comment:
If this https://github.com/quarkusio/quarkus/pull/3545 lands, we may need a link to it here.

Reply:
So the cited PR has landed, maybe we can add a reference to `How to Generate a Jandex Index` section?

Reply:
Yes please do that.

Reply:
Done. If the phrasing is OK I can update the hibernate guide with the same one.

Reply:
@gsmet is the new phrasing OK? Should I update the hibernate guide with it ?